tests/lib/cla_check.py

```diff
@@ -18,12 +18,35 @@
 shortlog_email_rx = re.compile("^\s*\d+\s+.*<(\S+)>$", re.M)
 
 
+is_travis = os.getenv("TRAVIS", "") == "true"
+is_github_actions = os.getenv("GITHUB_ACTIONS", "") == "true"
+
+
+def get_commit_range():
+    # Sanity check to ensure we are running from a pull request check job
+    if is_travis:
+        if os.getenv("TRAVIS_PULL_REQUEST", "false") == "false":
+            raise RuntimeError("called from a non-pull request Travis job")
+    elif is_github_actions:
+        if os.getenv("GITHUB_EVENT_NAME", "") != "pull_request":
+            raise RuntimeError("called from a non-pull request Github Actions job")
+    else:
+        raise RuntimeError("Unknown CI system.")
+
+    # The head revision is a synthesised merge commit, merging the
+    # proposed branch into the destination branch.  So the first
+    # parent is our destination, and the second is our proposal.
+    dest = check_output(["git", "rev-parse", "@^1"]).strip()
```

Comment:
We could check to make sure that the top commit is a merge (git cat-file -p | grep '^parent ' | wc -l > 1) or leave a comment that rev-parse fails when asking for a parent which isn't there.

Reply:
Good idea.  This also gives me the parent revision IDs directly, so can replace the rev-parse calls.